man/faq/selection-context.Rmd

```diff
@@ -20,6 +16,24 @@ mtcars[contains("foo")]
 subset(mtcars, select = matches("foo"))
 ```
 
-If you see this error, you've probably used a selection helper in the
+If you see this error, you may have used a selection helper in the
 wrong place, possibly as the result of a typo (e.g. misplaced comma or
-wrong argument name).
+wrong argument name). Alternatively, you may be deliberately trying
+to reduce duplication in your code by extracting out a selection into
+a variable:
+
+```{r, error = TRUE}
+my_vars <- c(name, species, ends_with("color"))
+```
+
+To make this work you'll need to do two things:
+
+* Wrap the whole thing in a function
+* Use `any_of()` or `all_of()` instead of bare variable names
```

Comment:
Does `all_of()` now return a vector of positions after the recent changes? If yes, do we have a unit test for the combination of `all_of()` and helpers like `starts_with()`? If not, we should keep #270 around, or open a new issue.

Reply:
Should we add an info bullet pointing to this page in the out-of-context error?

Reply:
I think nothing has changed with `all_of()`, except that (e.g) `all_of(1)` is now coerced to an integer vector.

And this is the Rd topic that `peek_vars()` points to 😄